samples/python/kinetics/custom_reactions.py

```diff
@@ -25,41 +25,67 @@
 custom_reactions = [r for r in reactions]
 custom_reactions[2] = ct.Reaction(
     equation='H2 + O <=> H + OH',
-    rate=lambda T: 38.7 * T**2.7 * exp(-3150.15428/T))
+    rate=lambda T: 38.7 * T**2.7 * exp(-3150.1542797022735/T))
+custom_reactions[4] = ct.Reaction(
+    equation='H2O2 + O <=> HO2 + OH',
+    rate=lambda T: 9630.0 * T**2.0 * exp(-2012.8781339950629/T))
 
 gas1 = ct.Solution(thermo='ideal-gas', kinetics='gas',
                    species=species, reactions=custom_reactions)
 
 # construct reactions based on ExtensibleRate: replace 2nd reaction with equivalent
 # ExtensibleRate
-@ct.extension(name="extensible-Arrhenius")
+class ExtensibleArrheniusData(ct.ExtensibleRateData):
+    __slots__ = ("T",)
+    def __init__(self):
+        self.T = None
+
+    def update(self, gas):
+        T = gas.T
+        if self.T != T:
+            self.T = T
+            return True
+        else:
+            return False
```

Comment:
I've updated this so more information is available at the point where the exception is thrown (so the class and method name can be identified). The output is now something like:
```
*******************************************************************************
CanteraError thrown by CVodesIntegrator::integrate:
CVodes error encountered. Error code: -9
The right-hand side routine failed at the first call.

Exceptions caught during RHS evaluation:

*******************************************************************************
CanteraError thrown by Lambda generated by Delegator::makeDelegate:
Method 'update' of class 'ExtensibleArrheniusData' did not return a value of type 'double'.
*******************************************************************************

Components with largest weighted error estimates:
0: -1.3642641433719054e+86
38: -10.117372889372218
7: 0.5206771669918789
8: -0.33262364359287805
18: -0.27521731705348845
5: 0.18190017240387882
17: 0.17516792601392808
39: -0.014443245050882536
3: 0.003992263897614869
9: 0.0023360039408578046
*******************************************************************************
```
I think the reference to `CVodesIntegrator::integrate` should be expected because `CVodesIntegrator` does some of its own error handling and not every exception raised during the integration necessarily has to propagate back to the user.